articles/rules/current/context.md

```diff
@@ -23,35 +23,49 @@ The following properties are available for the `context` object.
 | `clientName` | The name of the application (as defined on the dashboard). |
 | `clientMetadata` | An object for holding other application properties. Its keys and values are strings. |
+| `connectionID` | A string containing the connection's unique identifier |
 | `connection` | The name of the connection used to authenticate the user (such as: `twitter` or `some-google-apps-domain`) |
 | `connectionStrategy` | The type of connection. For social connection `connectionStrategy` === `connection`. For enterprise connections, the strategy will be `waad` (Windows Azure AD), `ad` (Active Directory/LDAP), `auth0` (database connections), and so on. |
+| `connectionOptions` | An object representing the options defined on the connection. `connectionOptions.tenant_domain` is a string containing the domain being used for authentication when using an Enterprise connection. `connectionOptions.domain_aliases` is an array containing the optional domains registered as aliases in addition to the primary domain (specified in the `connectionOptions.tenant_domain` property). |
+| `connectionMetadata` | An object representing metadata defined on the connection. Its keys and values are strings. |
 | `samlConfiguration` | An object that controls the behavior of the SAML and WS-Fed endpoints. Useful for advanced claims mapping and token enrichment (only available for `samlp` and `wsfed` protocol). |
 | `protocol` | <%= include('./_context-protocol.md') %> |
 | `stats` | An object containing specific user stats, like `stats.loginsCount`. Note that any of the counter variables returned as part of the `stats` object does not increase during [silent authentication](/api-auth/tutorials/silent-authentication) (as when `prompt=none`). |
 | `sso` | <%= include('./_context-sso.md') %> |
 | `accessToken` | Used to add custom namespaced claims to the [Access Token](/tokens/access-token). |
 | `idToken` | Used to add custom namespaced claims to the [ID Token](/tokens/id-token). |
+| `original_protocol` | After a [redirect rule](/rules/current/redirect) has executed and the authentication transaction is resumed, this property will be populated with the original protocol used to initiate the transaction. |
+| `multifactor` | An object representing the multifactor settings used in [implementing contextual MFA](/multifactor-authentication/custom). |
+| `redirect` | The object used to [implement the redirection of a user from a rule](/rules/current/redirect#how-to-implement-a-redirect). |
 | `sessionID` | Unique id for the authentication session. Value is kept only if `prompt=none`. |
 | `request` | <%= include('./_context-request.md') %> |
 
 ## Sample contents
 
 ```js
 {
+  tenant: 'mydomain',
   clientID: 'q2hn...pXmTUA',
   clientName: 'Default App',
   clientMetadata: {},
+  connectionID: 'con_V7s88...lgW97',
   connection: 'Username-Password-Authentication',
   connectionStrategy: 'auth0',
+  connectionOptions: {},
```

Comment:
@solepano my concern is that we would have to change the current example since it's not an enterprise connection. I've documented them at the top of the document as an alternative: https://github.com/auth0/docs/pull/6591/files#diff-8800034d7401d684ab84602473faf32aR33

Reply:
ok, can we do the same for all other objects?

Reply:
also, do we know which fields are readable vs writable? something that jared pointed out here and might be interesting to know https://auth0.slack.com/archives/CBTDY4DJN/p1533321091000141?thread_ts=1533304955.000071&cid=CBTDY4DJN

Reply:
and finally, why there's no review triggered from this PR? it would be easier to see the changes live. Maybe because you used a branch?

Reply:
> why there's no review triggered from this PR?

I looked through the last 5-10 PRs and it seems that reviews are no longer generated :(

> ok, can we do the same for all other objects?

Sorry, could you clarify what you mean? Which objects are you referring to and what sort of changes?

> also, do we know which fields are readable vs writable?

I'd have to dig into this and get an understanding of what Jared means by readable vs writeable since they all vary depending on the property. i.e.: `context.redirect` is writeable and so is `context.connectionMetadata` but it won't persist beyond the Rules lifetime. It might make more sense to break that out into it's own story and investigate to avoid holding up the release of this specific feature.

@solepano

Reply:
> It might make more sense to break that out into it's own story and investigate to avoid holding up the release of this specific feature.
Completely agree. @FadyMak went beyond the call of duty for this particular PR. we could add another story to backlog if we feel this "readable vs writable" distinction is necessary.